src/anyp/Uri.cc

```diff
@@ -937,55 +947,62 @@ AnyP::Uri::Uri(AnyP::UriScheme const &aScheme) :
 }
 
 // TODO: fix code duplication with AnyP::Uri::parse()
-char *
-AnyP::Uri::cleanup(const char *uri)
+SBuf
+AnyP::Uri::Cleanup(SBuf &uri)
```

Comment:
We should either clean the parameter (and return nothing) or return the cleaned URI (and keep the given URI constant). I suspect the latter is better, but I have not carefully studied this choice.
```suggestion
AnyP::Uri::Cleanup(const SBuf &uri)
```

Reply:
It was a non-const parameter because c_str() cannot be used on a const. Using a temporary local now.